src/librustc/hir/lowering.rs

```diff
@@ -5551,13 +5558,21 @@ impl<'a> LoweringContext<'a> {
         //     }
         // }
         if !self.is_async_body {
-            span_err!(
+            let mut err = struct_span_err!(
                 self.sess,
                 await_span,
                 E0728,
                 "`await` is only allowed inside `async` functions and blocks"
             );
-            self.sess.abort_if_errors();
+            err.span_label(await_span, "only allowed inside `async` functions and blocks");
```

Comment:
this seems redundant with the message above

Reply:
I'm trying to remove all primary spans without labels from the cli output. The main msg should be generic and the label text more targeted and suitable to show inline in (for example) VSCode.

Would you be ok if the main message was "`await` in a function or block not marked with `async`" or similar?

Reply:
I don't think we should have two messages conveying the same information-- the wording isn't the problem.

Reply:
Some people ignore the error message entirely and focus exclusively on the spans. I agree that unnecessary text is suboptimal, but I do not 

```
error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-51751.rs:11:20
   |
LL |     let finished = result.await;
   |                    ^^^^^^^^^^^^
```

vs

```
error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-51751.rs:11:20
   |
LL |     let finished = result.await;
   |                    ^^^^^^^^^^^^ only allowed inside `async` functions and blocks
```

vs

```
error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-51751.rs:11:20
   |
LL | fn main() {
   |    ---- this is not `async`
LL |     let result = inc(10000);
LL |     let finished = result.await;
   |                    ^^^^^^^^^^^^ only allowed inside `async` functions and blocks
```

vs 

```
error[E0728]: `await` is only allowed inside `async` functions and blocks
  --> $DIR/issue-51751.rs:11:20
   |
LL | fn main() {
   |    ---- this is not `async`
LL |     let result = inc(10000);
LL |     let finished = result.await;
   |                    ^^^^^^^^^^^^
```

The last case is what this would be like if we remove the label, which has some text in the secondary label, but no text on the primary label, which I feel is suboptimal.

Reply:
@estebank Hmm, ok -- leave it as is then; in context I think your solution is the best UX.